components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -56,6 +60,27 @@ CurlDownloadHandler::CurlDownloadHandler(
         m_http_headers.append("Cache-Control: no-cache");
         m_http_headers.append("Pragma: no-cache");
     }
+    std::set<std::string> const kReservedHeaders = {"range", "cache-control", "pragma"};
```

Comment:
I would prefer to have a `std::array` of `std::string_view` which contains these three reserved keys. The function should then traverse the array and throw exceptions if any of the key in the array appears.
Benefits:
1. Using `std::array` with `std::string_view` can be declared as `constexpr`, which is resolved in compile time
2. Using `std::string_view` to avoid run-time string allocations.

Note: same as `std::map` vs. `std::unordered_map`, if a set is really needed, we should use `unordered_set` if the order doesn't matter.

Reply:
fixed in next commit